src/main/java/org/jabref/logic/shared/DBMSSynchronizer.java

```diff
@@ -192,7 +192,10 @@ public void synchronizeLocalDatabase() {
                             localEntry.getSharedBibEntryData()
                                       .setVersion(sharedEntry.get().getSharedBibEntryData().getVersion());
                             for (Field field : sharedEntry.get().getFields()) {
-                                localEntry.setField(field, sharedEntry.get().getField(field), EntryEventSource.SHARED);
+                                Optional<String> sharedFieldValue = sharedEntry.get().getField(field);
+                                if (sharedFieldValue.isPresent()) {
```

Comment:
This should be a good use case for ifPresent lambda, I guess.

Reply:
sharedEntry.get().getField(field).ifPresent((v) -> localEntry.setField)

Reply:
Thank you for hint. I fixed a modification of an unmodifiable collection. Seems that this feature is used very seldom. Hope, it gets better.